deploy.sh

```diff
@@ -49,6 +49,24 @@
 set -e # exit immediately if any command exits with a non-zero status
 # https://fvue.nl/wiki/Bash:_Error_handling
 
+
+# Check if user is authenticated with Google Cloud before running deployment
+if command -v gcloud &> /dev/null ; then
+  ACTIVE_ACCOUNT=$(gcloud auth list --format="value(account)" --filter="status:ACTIVE")
+  if [ -z "$ACTIVE_ACCOUNT" ]; then
+    echo "You are not authenticated with Google Cloud. Press any key to authenticate..."
```

Comment:
So i noticed that sometimes even when account is active it will still prompt to re-authenticate. Something like this should cover that scenario. Hopefully it's not too brittle
```suggestion
  if [ "$(gcloud auth print-access-token 2>/dev/null | wc -c)" -le 200 ]; then
    echo "You are not authenticated with Google Cloud. Press any key to authenticate... (browser window may open)"
```

Reply:
fine with that but lets keep more deploy.sh refactors for later please, would like to get this FF out.